hedera-node/cli-clients/src/main/java/com/hedera/services/cli/signedstate/DumpContractStoresSubcommand.java

```diff
@@ -167,28 +196,125 @@ public static ContractKeyLocal from(ContractKey ckey) {
-                        final var contractKey = entry.left();
+                        final var contractKey = ContractKeyLocal.from(entry.left());
                         final var iterableContractValue = entry.right();
-                        visitor.accept(contractKey, iterableContractValue);
+                        nSlotsSeen.incrementAndGet();
+                        visitor.accept(contractKey, iterableContractValue.asUInt256());
                     },
                     THREAD_COUNT);
         } catch (final InterruptedException ex) {
             Thread.currentThread().interrupt();
             didRunToCompletion = false;
         }
 
+        System.err.printf(
+                "iterateThroughContractStorage: end - %d slots seen, completed? %b%n",
+                nSlotsSeen.get(), didRunToCompletion);
+
         return didRunToCompletion;
     }
 
+    boolean iterateThroughMigratedContractStorage(BiConsumer<ContractKeyLocal, UInt256> visitor) {
+
+        System.err.printf("iterateThroughMigratedContractStorage: start%n");
+
+        final var contractStorageStore = getMigratedContractStore();
+
+        System.err.printf("iterateThroughMigratedContractStorage: migrated%n");
+
+        final var nSlotsSeen = new AtomicLong();
+
+        if (contractStorageStore == null) return false;
+        contractStorageStore.keys().forEachRemaining(key -> {
+            // (Not sure how many temporary _copies_ of a byte arrays are made here ... best not to ask ...)
+            final var contractKeyLocal = ContractKeyLocal.from(
+                    new ContractKey(key.contractNumber(), key.key().toByteArray()));
+            final var slotValue = contractStorageStore.get(key);
+            assert (slotValue != null);
+            final var value = uint256FromByteArray(slotValue.value().toByteArray());
+            nSlotsSeen.incrementAndGet();
+            visitor.accept(contractKeyLocal, value);
+        });
+
+        System.err.printf("iterateThroughContractStorage: end - %d slots seen, completed? true%n", nSlotsSeen.get());
+
+        return true;
+    }
+
+    /** First migrates the contract store from the mono-service representation to the modular-service representations,
+     *  and then returns all contracts with bytecodes from the migrated contract store, plus the ids of contracts with
+     *  0-length bytecodes.
+     */
+    @SuppressWarnings("unchecked")
+    @Nullable
+    WritableKVState<SlotKey, SlotValue> getMigratedContractStore() {
+
+        final var fromStore = state.getRawContractStorage();
+        final var expectedNumberOfSlots = Math.toIntExact(fromStore.size());
+
+        // Start the migration with a clean, writable KV store.  Using the in-memory store here.
+
+        final var contractSchema = new ContractSchema();
+        final var contractSchemas = contractSchema.statesToCreate();
+        final StateDefinition<SlotKey, SlotValue> contractStoreStateDefinition = contractSchemas.stream()
```

Comment:
So I'm looking at, for example, classes called `WritableContractStateStore` and thinking that "store" is what we call the piece of the entire hedera state that belongs to one of our services (semantically).

However I also see I dropped the ball here by calling it a `contractStoreState...` which is backwards from `contractStateStore...` which is embarrassing given that I like to think I'm careful about naming things ... sorry!

Anyway ... I think you're saying that se use "store" and "state" at different levels of abstraction - "store" being a higher level than "state" - and since I seem to be working here at the "state" level I should be consistent with that?

Reply:
> Anyway ... I think you're saying that se use "store" and "state" at different levels of abstraction - "store" being a higher level than "state" - and since I seem to be working here at the "state" level I should be consistent with that?

Correct

Reply:
So I will distinguish _contract storage_ from _store_ and _state.  `ServicesState` calls the part of the tree that holds all the K/V slots for all contracts the _contract storage_.